src/app/hooks/useClickTracker/index.jsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useRef, useContext, useState, useCallback } from 'react';
+import { sendEventBeacon } from '#containers/ATIAnalytics/beacon/index';
+import { getComponentInfo } from '#app/lib/analyticsUtils/index';
+import { ServiceContext } from '#contexts/ServiceContext';
+import { RequestContext } from '#contexts/RequestContext';
+import { buildATIClickParams } from '#containers/ATIAnalytics/params';
+import { isValidClick } from './helpers';
+
+const useClickTracker = ({ pageData, componentName }) => {
+  const [hasBeenClicked, setHasBeenClicked] = useState(false);
+
+  const requestContext = useContext(RequestContext);
+  const serviceContext = useContext(ServiceContext);
+  const { service } = serviceContext;
+  const clickRef = useRef(null);
+
+  const eventTrackingProps = buildATIClickParams(
+    pageData,
+    requestContext,
+    serviceContext,
+  );
+
+  const handleClick = useCallback(
+    event => {
+      if (!hasBeenClicked && isValidClick(event)) {
+        setHasBeenClicked(true);
+        const componentInfo = getComponentInfo({
+          result: event.target.href || window.location.href,
+          componentName,
+          componentData: {
+            actionLabel: 'click',
+            child: event.target.tagName,
```

Comment:
has this been verified as correct with @JonBeeb , @jroebu14  wasn't sure what to provide here on his viewing tracking PR?

Reply:
These haven't been verified as such yet, they're mostly placeholders. That being said, I think they're appropriate values, if a user clicks on a brand by clicking the svg, it'll send `brand-click` to ati, also specifying that the svg was the child element that was actually clicked on.